api/main.py

```diff
@@ -286,16 +286,22 @@ async def put_node(node_id: str, node: Node, token: str = Depends(get_user)):
     """Update an already added node"""
     try:
         node.id = ObjectId(node_id)
-        node_from_id = await get_node(node.id)
-        is_valid, message = node_from_id.validate_node_transition(node.state)
+        node_from_id = await db.find_by_id(Node, node_id)
+        if not node_from_id:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail=f"Node not found with id: {node.id}"
```

Comment:
I think we should use here HTTP 404. Even the detailed message says that node hasn't been found.

Reply:
Definitely. It was a blunder.